File: 2026-1/sesion-de-clase-24/net/sesion-de-clase-24/Ejercicio3/Identificador.cs
using System;
using System.Threading;

namespace Ejercicio3;

public static class Identificador
{
    private static readonly object lockObject = new object();

    public static int Id { get; set; }

    public static void Incrementar()
    {
        lock (lockObject)
        {
            Id++;
            Console.WriteLine($"{Id}: producido por -> {Thread.CurrentThread.Name}");
        }
    }

    public static int ObtenerId()
    {
        lock (lockObject)
        {
            return Id;
        }
    }

    public static void ImprimirIdActualConsumido(string nombreConsumidor)
    {
        lock (lockObject)
        {
            Console.WriteLine($"{nombreConsumidor} leyo identificador actual: {Id}");
        }
    }
}

File: 2026-1/sesion-de-clase-24/java/src/main/java/ejercicio3/Identificador.java
package ejercicio3;


public class Identificador {
    private static int id = 0;
//    private static final Object lock = new Object();

    // Se establece un area critica
    // Se establece un MUTEX o exclusion mutua
    public synchronized static void incrementar() {
//        synchronized (lock) {
//            id++;
//            System.out.println(id + ": producido por -> " + Thread.currentThread().getName());
//        }
        id++;
        System.out.println(id + ": producido por -> " + Thread.currentThread().getName());
    }

    public synchronized static int obtenerId() {
        return id;
    }

    public synchronized static void imprimirIdActualConsumido(String nombreConsumidor) {
        System.out.println(nombreConsumidor + " leyo identificador actual: " + id);
    }
}